openedx/features/course_duration_limits/config.py

```diff
@@ -50,3 +51,8 @@ def set_value_for_content_type_gating_holdback(sender, event=None, user=None, **
                 )
             except (ExperimentKeyValue.DoesNotExist, AttributeError):
                 pass
+            except IntegrityError:
```

Comment:
rather than doing this check we could probably use get_or_create above?

Reply:
I thought about doing this as well, but `get_or_create` doesn't work for 2 reasons:
1. `get_or_create` does a similar "check if exists then create". It does make the time between the check and creation shorter though.
2. We need to create with the value, but we don't want to check with the value. We could have the first enrollment create a value="True" and the second try to create one with value="False" which would still cause an integrity error. Again, lessening the odds, but not a fix.

Reply:
i think get_or_create would be a transaction in platform but good point regarding the value. approved